src/dataset-builder/DomainCriteriaSelector.test.ts

```diff
@@ -32,15 +33,15 @@ describe('DomainCriteriaSelector', () => {
 
   it('renders the domain criteria selector', async () => {
     // Arrange
-    render(h(DomainCriteriaSelector, { state, onStateChange: jest.fn() }));
+    render(h(DomainCriteriaSelector, { state, onStateChange: jest.fn(), datasetId }));
```

Comment:
This test now passes in a value for `datasetId`. Is this state validated in the test? That is, does/can the test somehow assert that the `datasetId` was stored and used as expected?